source/math/Quaternion.ooc

```diff
@@ -243,6 +243,24 @@ Quaternion: cover {
 		result := [this w, this x, this y, this z]
 		result
 	}
+	toFloatTransform2D: func -> FloatTransform2D {
+		normalized := this normalized
+		
+		(nw, nx, ny, nz) := (normalized w, normalized x, normalized y, normalized z)
+		(nwSquared, nxSquared, nySquared, nzSquared) := (nw squared(), nx squared(), ny squared(), nz squared())
+		
+		a := 1.0f - 2.0f * (nySquared + nzSquared)
+		b := 2.0f * (nx * ny + nz * nw)
+		c := 2.0f * (nx * nz - ny * nw)
+		d := 2.0f * (nx * ny - nz * nw)
+		e := 1.0f - 2.0f * (nxSquared + nzSquared)
+		f := 2.0f * (ny * nz + nx * nw)
+		g := 2.0f * (nx * nz + ny * nw)
+		h := 2.0f * (ny * nz - nx * nw)
+		i := 1.0f - 2.0f * (nxSquared + nySquared)
```

Comment:
I changed my mind, sorry:
```ooc		
		a := 1.0f - 2.0f * (ny squared() + nz squared())
		b := 2.0f * (nx * ny + nz * nw)
		c := 2.0f * (nx * nz - ny * nw)
		d := 2.0f * (nx * ny - nz * nw)
		e := 1.0f - 2.0f * (nx squared() + nz squared())
		f := 2.0f * (ny * nz + nx * nw)
		g := 2.0f * (nx * nz + ny * nw)
		h := 2.0f * (ny * nz - nx * nw)
		i := 1.0f - 2.0f * (nx squared() + ny squared())
```
You only used the `[xyz]Squared` variables twice anyway, and they're cheap to compute. `nwSquared` wasn't actually used at all. There was a warning for that, but `test.sh` compiles silently, so we missed that.